vpc/service/gc_v3.go

```diff
@@ -508,25 +511,29 @@ ORDER BY RANDOM()
 	return nil
 }
 
-func (vpcService *vpcService) describeCollector(ctx context.Context, eniWQ chan string, describeWQ chan []string) {
+func (vpcService *vpcService) describeCollector(ctx context.Context, eniWQ chan string, describeWQ chan []string) error {
 	defer close(describeWQ)
 	enis := []string{}
 	for {
 		select {
 		case <-ctx.Done():
-			return
+			return errors.Wrap(ctx.Err(), "Context error while waiting on eniWQ")
 		case eni, ok := <-eniWQ:
 			if !ok {
 				// The queue is complete
 				if len(enis) > 0 {
 					describeWQ <- enis
```

Comment:
it looks like you also need the conditional channel write here to avoid deadlocking in this case as well